vendor/sigs.k8s.io/controller-reconciler/pkg/reconciler/manager/gcp/redis/manager.go

```diff
@@ -256,3 +261,56 @@ func NewService(ctx context.Context) (*redis.Service, error) {
 	client.UserAgent = UserAgent
 	return client, nil
 }
+
+// --------------------- Observables -------------------------------
+
+// Observables - i
+type Observables struct {
+	observables []reconciler.Observable
+	labels      reconciler.KVMap
+}
+
+// NewObservables - observables
+func NewObservables() *Observables {
+	return &Observables{
+		observables: []reconciler.Observable{},
+	}
+}
+
+// WithLabels - inject labels
+func (o *Observables) WithLabels(labels reconciler.KVMap) *Observables {
+	o.labels = labels
+	return o
+}
+
+// For - add
+func (o *Observables) For(parent , instanceid string) *Observables {
+	rm, err := NewRsrcManager(context.TODO(), "redis")
```

Comment:
This is not correct imho.
We are trying to provide and overvable interface that the reconciler should call.

Refer Disk manager implementation: func (rm *RsrcManager) Observe(observables ...reconciler.Observable) ([]reconciler.Object, error) {